valohai_yaml/objs/parameter_widget.py

```diff
@@ -0,0 +1,39 @@
+from typing import Optional, Union
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.types import SerializedDict
+
+
+class ParameterWidget(Item):
+    """An UI widget for editing parameter values."""
+
+    def __init__(
+        self,
+        *,
+        type: str,
+        settings: Optional[dict] = None,
+    ) -> None:
+        self.type = type.lower()
+        self.settings = settings
+
+    def serialize(self) -> Optional[Union[SerializedDict, str]]:
+        if self.settings:
+            return {
+                'type': self.type,
+                'settings': self.settings,
+            }
+        return self.type
+
+    @classmethod
+    def parse(cls, data: Union['ParameterWidget', dict, str]) -> 'ParameterWidget':
+        if isinstance(data, dict):
+            return cls(
+                type=data.get('type', ''),
+                settings=data.get('settings'),
+            )
+
+        if isinstance(data, cls):
+            return data
+
+        value = str(data).lower()
```

Comment:
What's the reason for the lower-casing? And why isn't it done for the other style of `type`?

Reply:
Not doing it for the dict was my error. Pushed a fix now.

The reasoning for lowercase was that people will use `SQL`, `Sql` and `sql` and I don't see the upside of making this case sensitive.

Reply:
Fair enough.